src/modules/form-dyn-messages.js

```diff
@@ -0,0 +1,78 @@
+(function ($) {
+
+	'use strict';
+
+	var html = $('html');
+
+	var sels = {
+		ctn: '.js-form-dyn-messages-ctn',
+		message: '.js-form-dyn-message',
+		template: '.js-form-dyn-message-template'
+	};
+
+	var resetMessage = function (msg) {
+		if (!!msg.length) {
+			App.modules.notify('changeState.update', {
+				item: msg,
+				state: 'visible',
+				action: 'off'
+			});
+			msg.empty();
+		}
+	};
+
+	var showMessage = function (form, fx) {
+		var ctn = form.find(sels.ctn);
+		var template = ctn.find(sels.template).filter('[data-fx="' + fx + '"]');
+		var msg = ctn.find(sels.message);
+
+		if (!!msg.length) {
+			resetMessage(msg);
+			msg.append($(template.html()));
+			msg.outerHeight();
+			App.modules.notify('changeState.update', {
+				item: msg,
+				state: 'visible',
+				action: 'on'
+			});
+			html.velocity('scroll', {
```

Comment:
If I'm correct, this is used to scroll back up in case the message is higher on the screen. I think this is a patch for a UX problem that should be fixed by the design and front-end thus removing the need for this scroll.

Reply:
How do you want to handle it ? On a case by case basis ?

Reply:
> questionable design decision that led to the necessity of the scroll on a project

We need to have tools to deal with those ;)

> as an option we put on the form?

I have not problems with that. Both js and data-attr.